src/MasterXaction.h

```diff
@@ -40,21 +43,35 @@ class MasterXaction : public RefCountable
 {
 public:
     typedef RefCount<MasterXaction> Pointer;
+    explicit MasterXaction(const XactionInitiator);
+
+    MasterXaction(const XactionInitiator, ConnStateData *);
+
+    MasterXaction(const XactionInitiator, Comm::ConnectionPointer);
```

Comment:
> The idea is to let the user a choice of two options: pass either manager or connection, but not both.

I understand the rationale now, thank you. To be more precise, we are forcing a choice among three (not two) options: manager, connection, or nothing. I think this API is acceptable and possibly even the best (within this PR scope). Since those that poses ConnStateData can use two constructors, let's codify the rule:

```
    // Creators must call the first constructor they can call (with a non-nil
    // pointer). Following this rule maximizes stored information.
    MasterXaction(const XactionInitiator, ConnStateData *);
    MasterXaction(const XactionInitiator, Comm::ConnectionPointer);    
    explicit MasterXaction(const XactionInitiator anInitiator) : initiator(anInitiator) {};
```